### PR TITLE
monitoring: access token

### PR DESCRIPTION
**Pull request by rerowep**
* Add monitoring user and access token for setup.

### CODE REVIEW

File: scripts/setup [hunk 1]
```diff
@@ -111,6 +111,21 @@ invenio access allow admin-access role moderator
...
+invenio users create -a monitoring@rero.ch --password monitor
+invenio users confirm monitoring@rero.ch
+invenio roles create monitoring
+invenio roles add monitoring@rero.ch monitoring
+# create token access for monitoring
+# if the environement variable INVENIO_RERO_ACCESS_TOKEN_MONITORING is not set
+# a new token will be generated
+if [ -z ${INVENIO_RERO_ACCESS_TOKEN_MONITORING} ]
+then
+    eval ${PREFIX} invenio utils tokens_create -n monitoring -u monitoring@rero.ch
```
Comment (jma) [inline scripts/setup:124]: You can create a variable with a base command.

File: sonar/modules/cli/utils.py [hunk 2]
```diff
@@ -178,7 +185,89 @@ def export(pid_type, serializer_key, output_dir):
...
 
-    except Exception as exception:
-        click.secho('An error occured during export: {error}'.format(
-            error=str(exception)),
-                    fg='red')
+    except Exception as err:
+        click.secho(f'An error occured during export: {err}', fg='red')
+
+
+def create_personal(
```
Comment (jma) [inline sonar/modules/cli/utils.py:192]: This can be a candidate for rero-invenio-base.

**Review by jma — APPROVED** [review]
Tests are missing.

File: sonar/modules/cli/utils.py [hunk 3]
```diff
@@ -178,7 +185,89 @@ def export(pid_type, serializer_key, output_dir):
...
+            is_personal=True,
+            is_internal=is_internal,
+        )
+        db.session.add(client)
+        db.session.add(token)
+
+    return token
+
+
+@utils.command('tokens_create')
```
Comment (Garfield-fr) [inline sonar/modules/cli/utils.py:239]: maybe `token_create` ?

File: sonar/modules/cli/utils.py [hunk 4]
```diff
@@ -178,7 +185,89 @@ def export(pid_type, serializer_key, output_dir):
...
+    '-u', '--user', required=True, callback=process_user,
+    help='User ID or email.')
+@click.option(
+    '-s', '--scope', 'scopes', multiple=True, callback=process_scopes)
+@click.option('-i', '--internal', is_flag=True)
+@click.option(
+    '-t', '--access_token', 'access_token', required=False,
+    help='personalized access_token.')
+@with_appcontext
+def tokens_create(name, user, scopes, internal, access_token):
```
Comment (Garfield-fr) [inline sonar/modules/cli/utils.py:251]: `token_create` ?

File: scripts/setup [hunk 5]
```diff
@@ -111,6 +111,21 @@ invenio access allow admin-access role moderator
 invenio access allow admin-access role submitter
 message "Done" "success"
 
+section "Create user for monitoring" "info"
+invenio users create -a monitoring@rero.ch --password monitor
+invenio users confirm monitoring@rero.ch
+invenio roles create monitoring
+invenio roles add monitoring@rero.ch monitoring
+# create token access for monitoring
+# if the environement variable INVENIO_RERO_ACCESS_TOKEN_MONITORING is not set
```
Comment (Garfield-fr) [inline scripts/setup:120]: `environement` => `environment`

File: scripts/setup (hunk 1 above)
Comment (Garfield-fr) [inline scripts/setup:124]: `tokens_create` => `token_create` If you change the function name below

File: scripts/setup [hunk 6]
```diff
@@ -111,6 +111,21 @@ invenio access allow admin-access role moderator
...
+# a new token will be generated
+if [ -z ${INVENIO_RERO_ACCESS_TOKEN_MONITORING} ]
+then
+    eval ${PREFIX} invenio utils tokens_create -n monitoring -u monitoring@rero.ch
+else
+    eval ${PREFIX} invenio utils tokens_create -n monitoring -u monitoring@rero.ch -t ${INVENIO_RERO_ACCESS_TOKEN_MONITORING}
```
Comment (Garfield-fr) [inline scripts/setup:126]: `tokens_create` => `token_create` If you change the function name below

**Review by Garfield-fr — APPROVED** [review]

File: sonar/modules/cli/utils.py (hunk 2 above)
Comment (rerowep) [reply sonar/modules/cli/utils.py:192]: Yes

**Review by Garfield-fr — APPROVED** [review]

**Review by PascalRepond — APPROVED** [review]

**Pull request merged**